src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -69,71 +62,66 @@ const ContentTagsDropDownSelector = ({
-                      tabIndex="0"
-                      onKeyPress={(event) => (event.key === 'Enter' ? clickAndEnterHandler(i) : null)}
-                    />
-                  </div>
-                )}
+    <> {/* <<<< TODO: make this into a <div> and put the left padding here so it affects everything - spinners, load more, tags, ... */}
+      {tagPages.map((tagPage, pageNum) => <React.Fragment key={pageNum}>
+        {tagPage.isLoading ? "Loading..." : null /* TODO: Use <LoadingSpinner /> from master src/components/generic */}
+        {tagPage.isError ? "Error..." : null /* TODO: show a proper error message */}
+        {tagPage.data?.map((tagData) =>
+          <React.Fragment key={tagData.value}>
+            <div
+              className="d-flex flex-row"
+              style={{
+                paddingLeft: `${level * 1}rem`,
+                minHeight: '44px',
+              }}
+            >
+              <div className="d-flex">
+                <SelectableBox
+                  inputHidden={false}
+                  type="checkbox"
+                  className="d-flex align-items-center taxonomy-tags-selectable-box"
+                  aria-label={`${tagData.value} checkbox`}
+                  data-selectable-box="taxonomy-tags"
+                  value={[...lineage, encodeURIComponent(tagData.value)].join(',')}
+                  isIndeterminate={isImplicit(tagData)}
+                  disabled={isImplicit(tagData)}
+                >
+                  {tagData.value}
+                </SelectableBox>
+                { tagData.childCount > 0
+                  && (
+                    <div className="d-flex align-items-center taxonomy-tags-arrow-drop-down">
+                      <Icon
+                        src={isOpen(tagData.value) ? ArrowDropUp : ArrowDropDown}
+                        onClick={() => clickAndEnterHandler(tagData.value)}
+                        tabIndex="0"
+                        onKeyPress={(event) => (event.key === 'Enter' ? clickAndEnterHandler(tagData.value) : null)}
+                      />
+                    </div>
+                  )}
+              </div>
+
             </div>
 
-          </div>
-
-          { taxonomyTag.subTagsUrl && isOpen(i) && (
-            <ContentTagsDropDownSelector
-              key={`selector-${taxonomyTag.value}`}
-              taxonomyId={taxonomyId}
-              subTagsUrl={taxonomyTag.subTagsUrl}
-              level={level + 1}
-              lineage={[...lineage, encodeURIComponent(taxonomyTag.value)]}
-              tagsTree={tagsTree}
-            />
-          )}
-        </>
-      ))}
-
-      { nextPage && isTaxonomyTagsLoaded
+            { tagData.childCount > 0 && isOpen(tagData.value) && (
+              <ContentTagsDropDownSelector
+                taxonomyId={taxonomyId}
+                level={level + 1}
+                lineage={[...lineage, encodeURIComponent(tagData.value)]}
```

Comment:
Nit: Why are you using `encodeURIComponent` with an array? I would think that with the array separating the string values, there is no need for encoding/decoding here.